automation/check-patch.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash -xe
+
+main() {
+    TARGET="$0"
+    TARGET="${TARGET#./}"
+    TARGET="${TARGET%.*}"
+    TARGET="${TARGET#*.}"
+    echo "TARGET=$TARGET"
+    export TARGET
+
+    echo "Run functional tests"
+    exec automation/test.sh
```

Comment:
why use `exec` here instead of running `automation/test.sh` in the same shell?

Reply:
To avoid the creation of a new process in the process tree